includes/Admin.php

```diff
@@ -96,57 +98,30 @@ public function prefill_post_content( $content ) {
 
 		$post_id = absint( sanitize_text_field( (string) wp_unslash( $_GET['from-web-story'] ) ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 
-		if ( ! $post_id || Story_Post_Type::POST_TYPE_SLUG !== get_post_type( $post_id ) ) {
+		if ( ! $post_id ) {
 			return $content;
 		}
 
 		if ( ! current_user_can( 'read_post', $post_id ) ) {
 			return $content;
 		}
 
-		$block_markup_with_poster = <<<BLOCK
-<!-- wp:web-stories/embed {"url":"%1\$s","title":"%2\$s","poster":"%3\$s"} -->
-<div class="wp-block-web-stories-embed alignnone">
-	<amp-story-player style="width:360px;height:600px" data-testid="amp-story-player"><a
-			href="%1\$s"
-			style="--story-player-poster:url('%3\$s')">%4\$s</a>
-	</amp-story-player>
-</div>
-<!-- /wp:web-stories/embed -->
-BLOCK;
-
-		$block_markup_without_poster = <<<BLOCK
-<!-- wp:web-stories/embed {"url":"%1\$s","title":"%2\$s","poster":""} -->
-<div class="wp-block-web-stories-embed alignnone">
-	<amp-story-player style="width:360px;height:600px" data-testid="amp-story-player"><a
-			href="%1\$s"
-			>%3\$s</a>
-	</amp-story-player>
-</div>
-<!-- /wp:web-stories/embed -->
-BLOCK;
-
-		$url        = (string) get_the_permalink( $post_id );
-		$title      = (string) get_the_title( $post_id );
-		$has_poster = has_post_thumbnail( $post_id );
-
-		if ( $has_poster ) {
-			$poster = (string) wp_get_attachment_image_url( (int) get_post_thumbnail_id( $post_id ), Media::POSTER_PORTRAIT_IMAGE_SIZE );
-
-			return sprintf(
-				$block_markup_with_poster,
-				esc_url( $url ),
-				esc_js( $title ),
-				esc_url( $poster ),
-				esc_html( $title )
-			);
+		$story = new Story();
+		if ( ! $story->load_from_post( $post_id ) ) {
+			return $content;
 		}
 
+		$renderer = new Embed( $story, 360, 600, 'none' );
+		$html     = $renderer->render();
+
+		$block_markup = '<!-- wp:web-stories/embed {"url":"%1$s","title":"%2$s","poster":"%3$s"} -->%4$s<!-- /wp:web-stories/embed -->';
+
 		return sprintf(
-			$block_markup_without_poster,
-			esc_url( $url ),
-			esc_js( $title ),
-			esc_html( $title )
+			$block_markup,
+			esc_url( $story->get_url() ),
+			esc_js( $story->get_title() ),
```

Comment:
Here we used `get_the_title` and all the filters that come with it, but now not anymore. Could that cause issues?

Reply:
So should I change it here https://github.com/google/web-stories-wp/pull/3733/files#diff-3193ce2922263d4f53be614e93cdae9aR103

Reply:
No, because we occasionally need the raw title. (Just scroll down to `\Google\Web_Stories\Admin::prefill_post_title`)

So we need a way to get the raw title, and a way to get the title via `get_the_title()`

The easiest solution here seems to be when `load_from_post()` stores the post object (or ID) as class property, and then we could do:

```php
public function get_title() {
if ( $this->post ) {
  return get_the_title( $this->post ); // ensures that all filters and translations will always be applied
}

return $this->get_raw_title();
}

public function get_raw_title() {
  return $this->title;
}
```

I know this kinda defeats the purpose of `load_from_post` but otherwise something like this won't work:

```php
$story = new Story();
$story->load_from_post( $post );
echo $story->get_title(); // e.g. 'Post Title'.

add_filter( 'the_title', '__return_empty_string' );

echo $story->get_title(); // Will still be 'Post Title', because filters are not running
```

Reply:
Just thinking out loud hear, so curious to hear your thoughts

Reply:
There is only one place that needed a raw title and I have now removed it.

Reply:
Fixed https://github.com/google/web-stories-wp/pull/3733/commits/a9f814eaff35bffe6c0ad0a08dfa5a27242d43a8